### PR TITLE
chore: add npm publish in release workflow

### PR DESCRIPTION
**Pull request by dzianis-dashkevich**
Note: `NPM_TOKEN` should be added as repo's secrete

### CODE REVIEW

File: COLLABORATOR_GUIDE.md [hunk 1]
```diff
@@ -3,11 +3,7 @@
...
 # Releasing
 
-- Ensure the tests are passing on master
+- Ensure the tests are passing on main
 - You may need to be granted npm publish permissions if this is your first time publishing. Try to get this working first!
 - Version on npm `npm version [patch|minor|major]`
-- Push changes from versioning to master, and the new tag you just created `git push && git push --tags`. You can also push the specific tag with `git push [remote] [tag]`
-- Publish on npm `npm publish`
-- Go to https://github.com/videojs/mux.js/releases
-- Click "Draft a new release"
```
Comment (wseymour15) [inline COLLABORATOR_GUIDE.md:12]: Does the draft still need to be manually released on Github? It don't see anything that would automate this, but I could be missing something.

File: COLLABORATOR_GUIDE.md [hunk 2]
```diff
@@ -3,11 +3,7 @@
...
-- Fill out the form with the released version and upload the files in the `/dist` directory
+- Push changes from versioning to main, and the new tag you just created `git push && git push --tags`. You can also push the specific tag with `git push [remote] [tag]`
```
Comment (gkatsev) [inline COLLABORATOR_GUIDE.md:9]: Best to use `git push --follow-tags origin main`, like VHS mentions too. It's a bit safer, too.

**Comment by gkatsev** [comment]
also, same comment as https://github.com/videojs/http-streaming/pull/1377#pullrequestreview-1318036306 heh

File: COLLABORATOR_GUIDE.md (hunk 1 above)
Comment (dzianis-dashkevich) [reply COLLABORATOR_GUIDE.md:12]: @wseymour15 release workflow will create GitHub release

**Review by wseymour15 — APPROVED** [review]
LGTM! +1 Once Gary's comments are resolved.

File: COLLABORATOR_GUIDE.md (hunk 2 above)
Comment (dzianis-dashkevich) [reply COLLABORATOR_GUIDE.md]: done

**Comment by dzianis-dashkevich** [comment]
@gkatsev updated

**Review by gkatsev — APPROVED** [review]

**Pull request merged**